src/freenet/client/filter/M3UFilter.java

```diff
@@ -116,65 +117,63 @@ public void readFilter(InputStream input, OutputStream output, String charset, H
+                                    String subMimetype = ContentFilter.mimeTypeForSrc(uriold);
+                                    // add prefix for the host name
+                                    // for absolute path names,
+                                    // because otherwise external
+                                    // clients could be tricked into
+                                    // accessing local files (and some
+                                    // just don't work, especially not
+                                    // with downloaded files). This
+                                    // can however make downloaded
+                                    // files leak information about
+                                    // the local setup (host and
+                                    // port).
 
-                                // mirroring tools like `wget -mk`
-                                // strip the absolute path again,
-                                // so mirroring should not be
-                                // impaired.
-                                if (cb instanceof GenericReadFilterCallback) {
-                                    try {
-                                        filtered = ((GenericReadFilterCallback)cb).makeURIAbsolute(filtered);
-                                    } catch (URISyntaxException e) {
-                                        filtered = badUriReplacement;
+                                    // mirroring tools like `wget -mk`
+                                    // strip the absolute path again,
+                                    // so mirroring should not be
+                                    // impaired.
+                                    filtered = cb.processURI(uriold, subMimetype, hostPort);
+                                    // allow transparent pass through
+                                    // for all but the largest files,
+                                    // but not for external
+                                    // links. This check is safe,
+                                    // since false positives will just
+                                    // lead to a file to not be played
+                                    // (players will get progress-bar
+                                    // HTML content instead).
+                                    if (!filtered.contains(ExternalLinkToadlet.PATH)
+                                        && !filtered.contains(ExternalLinkToadlet.magicHTTPEscapeString)) {
+                                        if (filtered.contains("?")) {
+                                            filtered += "&";
+                                        } else {
+                                            filtered += "?";
+                                        }
+                                        filtered += "max-size=" + Long.valueOf(
+                                            MAX_LENGTH_NO_PROGRESS).toString();
```

Comment:
```suggestion
                                        filtered += "max-size=" + String.valueOf(MAX_LENGTH_NO_PROGRESS);
```

Reply:
It’s even simpler: I can just leave out the explicit valueOf, since the long will be cast to a string anyway.